### PR TITLE
Add Child dental insurance route

### PR DESCRIPTION
**Pull request by shawn320**
### Description
dental-insurance screen in adult child flow. will need to refactor the button link when child-information and other screens are available

### Checklist

- [ ] I have tested the changes locally
- [ ] I have updated the documentation if necessary
- [ ] I have added/updated tests that prove my fix is effective or that my feature works
- [ ] I have checked that my code follows the project's coding style by running `npm run format:check`
- [ ] I have checked that my code contains no linting errors by running `npm run lint`
- [ ] I have checked that my code contains no type errors by running `npm run typecheck`
- [ ] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [ ] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

**Review by sebastien-comeau — CHANGES_REQUESTED** [review]
Looking at it quickly... `adult-child/dental-insurance` should not be moved since it's required for the main applicant (parent). Make a copy of it to `children/:childId/dental-insurance` and modified it to suit the single child flow.

**Comment by shawn320** [comment]
> Looking at it quickly... `adult-child/dental-insurance` should not be moved since it's required for the main applicant (parent). Make a copy of it to `children/:childId/dental-insurance` and modified it to suit the single child flow.

I thought I was working on child flow, my bad. I'll add the adult dental insurance back.

File: frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/children/$childId/dental-insurance.tsx [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+  pageIdentifier: pageIds.public.apply.adultChild.dentalInsurance,
+  pageTitleI18nKey: 'apply-adult-child:children.dental-insurance.title',
+};
+
+export const meta: MetaFunction<typeof loader> = mergeMeta(({ data }) => {
+  return data ? getTitleMetaTags(data.meta.title) : [];
+});
+
+export async function loader({ context: { session }, params, request }: LoaderFunctionArgs) {
+  const state = loadApplyAdultChildState({ params, request, session });
```
Comment (sebastien-comeau) [inline frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/children/$childId/dental-insurance.tsx:40]: Use `loadApplyAdultSingleChildState` instead.

```suggestion
  const state = loadApplyAdultSingleChildState({ params, request, session });
```

File: frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/children/$childId/dental-insurance.tsx [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+  const t = await getFixedT(request, handle.i18nNamespaces);
+  const child = state.children.find((child) => child.id === params.childId);
```
Comment (sebastien-comeau) [inline frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/children/$childId/dental-insurance.tsx:42]: Child state will be loaded from `loadApplyAdultSingleChildState`.

File: frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/children/$childId/dental-insurance.tsx [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+  const csrfToken = String(session.get('csrfToken'));
+  const meta = { title: t('gcweb:meta.title.template', { title: t('apply-adult-child:children.dental-insurance.title') }) };
+
+  return json({ id: state, csrfToken, meta, defaultState: state.dentalInsurance, childName, editMode: state.editMode });
+}
+
+export async function action({ context: { session }, params, request }: ActionFunctionArgs) {
+  const log = getLogger('apply/adult-child/dental-insurance');
+
+  const state = loadApplyAdultChildState({ params, request, session });
```
Comment (sebastien-comeau) [inline frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/children/$childId/dental-insurance.tsx:54]: Matches what is done in `information` route.

```suggestion
  const state = loadApplyAdultSingleChildState({ params, request, session });
  const applyState = loadApplyAdultChildState({ params, request, session });
```

File: frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/children/$childId/dental-insurance.tsx [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+  }
+
+  const dentalInsurance = formData.get('dentalInsurance') ? formData.get('dentalInsurance') === 'yes' : undefined;
+  const parsedDataResult = dentalInsuranceSchema.safeParse(dentalInsurance);
+
+  if (!parsedDataResult.success) {
+    return json({ errors: parsedDataResult.error.format()._errors });
+  }
+
+  saveApplyState({ params, session, state: { dentalInsurance: parsedDataResult.data } });
```
Comment (sebastien-comeau) [inline frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/children/$childId/dental-insurance.tsx:76]:
```suggestion
  saveApplyState({
    params,
    session,
    state: {
      children: applyState.children.map((child) => {
        if (child.id !== state.id) return child;
        return { ...child, dentalInsurance: parsedDataResult.data };
      }),
    },
  });
```

File: frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/children/$childId/dental-insurance.tsx [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+            </div>
+          ) : (
+            <div className="mt-8 flex flex-row-reverse flex-wrap items-center justify-end gap-3">
+              <Button variant="primary" data-gc-analytics-customclick="ESDC-EDSC:CDCP Online Application Form:Continue - Access to other dental insurance click">
+                {t('children.dental-insurance.button.continue')}
+                <FontAwesomeIcon icon={isSubmitting ? faSpinner : faChevronRight} className={cn('ms-3 block size-4', isSubmitting && 'animate-spin')} />
+              </Button>
+              <ButtonLink
+                id="back-button"
+                routeId="$lang+/_public+/apply+/$id+/adult-child/communication-preference"
```
Comment (sebastien-comeau) [inline frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/children/$childId/dental-insurance.tsx:201]: Should redirect to child information route.

File: frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/children/$childId/dental-insurance.tsx [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+    return json({ errors: parsedDataResult.error.format()._errors });
+  }
+
+  saveApplyState({ params, session, state: { dentalInsurance: parsedDataResult.data } });
+
+  if (state.editMode) {
+    return redirect(getPathById('$lang+/_public+/apply+/$id+/adult-child/review-information', params));
+  }
+
+  return redirect(getPathById('$lang+/_public+/apply+/$id+/adult-child/federal-provincial-territorial-benefits', params));
```
Comment (sebastien-comeau) [inline frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/children/$childId/dental-insurance.tsx:82]: Redirect to children HUB for now.

```suggestion
  return redirect(getPathById('$lang+/_public+/apply+/$id+/adult-child/children/index', params));
```

**Review by sebastien-comeau — CHANGES_REQUESTED** [review]

**Comment by shawn320** [comment]
@sebastien-comeau Question: How to properly show child name in the page title? It's currently showing {{childName}}

**Review by sebastien-comeau — APPROVED** [review]
Child's information should redirect that route after submission. Change it in another PR.

**Comment by sebastien-comeau** [comment]
> @sebastien-comeau Question: How to properly show child name in the page title? It's currently showing {{childName}}

```tsx
t('apply-adult-child:children.dental-insurance.title', { childName })
```
https://www.i18next.com/translation-function/interpolation

**Comment by shawn320** [comment]
> > @sebastien-comeau Question: How to properly show child name in the page title? It's currently showing {{childName}}
> 
> ```tsx
> t('apply-adult-child:children.dental-insurance.title', { childName })
> ```
> 
> https://www.i18next.com/translation-function/interpolation

Yes, but I'm not sure how to access state outside loader and action. The page title is in `handle`

**Review by sebastien-comeau — APPROVED** [review]

**Pull request merged**